src/middleware/dune

```diff
@@ -21,6 +24,19 @@
  (preprocess (pps lwt_ppx))
  (instrumentation (backend bisect_ppx)))
 
+(library
+ (public_name dream.middleware.static)
+ (name dream__static)
+ (modules static)
+ (libraries
+  digestif
+  dream.pure
+  lwt.unix
+  magic-mime
+ )
+ (preprocess (pps lwt_ppx))
+ (instrumentation (backend bisect_ppx)))
```

Comment:
Rather than doing it this way, I recommend to create `src/unix/`, inside it `dream.unix`, and move `Dream.static` middleware there. Then, we can put any other Unix-only things that arise over time into that directory &mdash; should be more future-proof.

Is there any analogue to `Dream.static` that would make sense for Mirage?

Reply:
> Is there any analogue to Dream.static that would make sense for Mirage?

MirageOS has nothing at the beginning and, according to your application, you can "plug" a file-system which should respect a certain interface. Currently, we have [`Mirage_kv.S`](https://github.com/mirage/mirage-kv) as a common interface to get some values from a key and [`ocaml-crunch`](https://github.com/mirage/ocaml-crunch) or [`docteur`](https://github.com/dinosaure/docteur) as a possible implementation of this specific interface.

It's what we can propose to the MirageOS user as a _read-only_ "file-system". However, such choice is owned by the end-user and we should keep the ability to make an operating system without a file-system and with `dream`. My idea is to let the user to use a `static` middleware from a `Mirage_kv.S` module if he wants - but `dream.mirage` should not integrate it as a _standalone_ middleware.

Reply:
Thanks for telling me about that, it makes sense, and I agree!